File: 658-find-k-closest-elements/main.go
func findClosestElements(arr []int, k int, x int) []int {
    
    left, right := 0, len(arr) - 1

    for right - left >= k {
        if abs(arr[left] - x) > abs(arr[right] - x) {
            left++
        }else {
            right--
        }
    }

    return arr[left: right + 1]
}

func abs(x int) int {
    if x < 0 {
        return -x
    }
    return x
}

File: 658-find-k-closest-elements/main.rs
impl Solution {
    pub fn find_closest_elements(arr: Vec<i32>, k: i32, x: i32) -> Vec<i32> {
        
        let mut left: usize = 0;
        let mut right = arr.len() - 1;

        while (right - left) as i32 >= k {
            if (arr[left] as i32 - x).abs() > (arr[right] as i32 - x).abs() {
                left += 1;
            }else {
                right -= 1;
            }
        }

        arr[left..right + 1].to_vec()
    }
}
